ios/Classes/FacebookModule.m

```diff
@@ -394,17 +322,19 @@ -(void)authorize:(id)args
 {
     ENSURE_SINGLE_ARG_OR_NIL(args, NSNumber);
     BOOL allowUI = args == nil ? YES : NO;
-
+    NSArray *permissions_ = permissions == nil ? [NSArray array] : permissions;
+    FBSDKLoginManager *loginManager = [[[FBSDKLoginManager alloc] init] autorelease];
     TiThreadPerformOnMainThread(^{
-        NSArray *permissions_ = permissions == nil ? [NSArray array] : permissions;
-        BOOL sessionOpened = [FBSession openActiveSessionWithReadPermissions: allowUI ? permissions_ : FBSession.activeSession.permissions allowLoginUI:allowUI completionHandler:
-                                  ^(FBSession *session,
-                                    FBSessionState state, NSError *error) {
-                                      [self sessionStateChanged:session state:state error:error];
-            }];
-            NSLog(@"[DEBUG] openActiveSessionWithReadPermissions returned: %d", sessionOpened);
-        
-    }, NO);
+        [loginManager logInWithReadPermissions: permissions_ fromViewController:nil handler:^(FBSDKLoginManagerLoginResult *result, NSError *error) {
+            if (error) {
+                DebugLog(@"[ERROR] Process error.");
+            } else if (result.isCancelled) {
+                DebugLog(@"[ERROR] User Cancelled");
+            } else {
+                DebugLog(@"[INFO] Logged in");
+            }
+        }];
```

Comment:
Should this logs remain there? If so, unify the sentences with a closing "." and check the lowercase spelling of "Cancelled".